test/PaymentRelay.test.ts

```diff
@@ -26,94 +27,241 @@ contract("PaymentRelay", (accounts) => {
+            from: sender,
+          });
+          assert.fail("execPayment() did not throw.");
         } catch (e: any) {
-          expect(e.message).to.includes("already processed");
+          expect(e.message).to.includes("missing role");
         }
       });
 
-      it("Should block payment to unauthorized receiver", async function () {
-        const ANY_UID = web3.utils.keccak256("ANY_UID");
+      it("Should block execution of payments that were not previously reserved", async function () {
+        try {
+          await paymentRelayContract.execPayment(sender, ANY_UID, receiver, {
+            from: sender,
+          });
+          assert.fail("execPayment() did not throw.");
+        } catch (e: any) {
+          expect(e.message).to.includes("Payment reserve not found");
+        }
+      });
+
+      it("Should block refund of payments that were not previously reserved", async function () {
+        try {
+          await paymentRelayContract.refundPayment(sender, ANY_UID, {
+            from: operator,
+          });
+          assert.fail("refundPayment() did not throw.");
+        } catch (e: any) {
+          expect(e.message).to.includes("Payment reserve not found");
+        }
+      });
 
+      it("Should only allow reservation owner, or contract operator to execute payment", async function () {
         try {
-          await champContract.approve(paymentRelayContract.address, amount, { from: sender });
           await paymentRelayContract.execPayment(
-            champContract.address,
-            amount,
-            ANY_UID,
-            anyUser,
-            { from: sender }
+            sender,
+            PAYMENT_UID,
+            receiver,
+            { from: anyUser }
           );
-          assert.fail("operatorSend() did not throw.");
+          assert.fail("execPayment() did not throw.");
         } catch (e: any) {
-          expect(e.message).to.includes("missing role");
+          expect(e.message).to.includes(
+            "Caller does not have permission to execute payment"
+          );
+        }
+      });
+
+      it("Should only allow contract operator to refund payment", async function () {
+        try {
+          await paymentRelayContract.refundPayment(sender, PAYMENT_UID, {
+            from: anyUser,
+          });
+          assert.fail("refundPayment() did not throw.");
+        } catch (e: any) {
+          expect(e.message).to.includes(
+            "Caller does not have permission to refund payment"
+          );
         }
+
+        try {
+          await paymentRelayContract.refundPayment(sender, PAYMENT_UID, {
+            from: sender,
+          });
+          assert.fail("refundPayment() did not throw.");
+        } catch (e: any) {
+          expect(e.message).to.includes(
+            "Caller does not have permission to refund payment"
+          );
+        }
+      });
+    });
+
+    describe("When payment is executed", function () {
```

Comment:
P1, WDYT? 

Missing cases:
- should prevent to refund payment
- should prevent to execute payment again
- should prevent reserve payment with the same UID